### DISCUSSION TITLE
Bug : Github Copilot suggestion is out of topic in that case…

### DISCUSSION THREAD

**Discussion by ytrezq** [community/community#19393] [Copilot Conversations]
There are 2 types of cryptocurrency : real cryptocurrency and tokens which use the mining network of a compatible cryptocurrency.
In the case of Ethereum, they are called ERC20. They are simply regular programs called smart‑contracts which belong to an account that in the case of cryptocurrency are addresses.
GitHub copilot has no problems generating code for fetching token price by name. But with Ethereum, you can have several copies of the same token. So the proper way to specify a token is to do it by it’s ɪᴅ : I mean the unique Ethereum account the token belongs to.

So it’s about getting cryptocurrency price by address instead of by name :

    """fetch ERC20 token price in Ether from address
    input:
       address: Ethereum token contract's address
    Output:
       wei: ETher price
    """
and GitHub copilot propose :

    def fetch_erc20_price(address):
        import requests
        import json
        import time
        import sys
        import os
        import web3
        import web3.middleware.geth_poa_middleware
        import eth_utils
        from web3 import Web3, HTTPProvider
        from web3.middleware.geth_poa_middleware import geth_poa_gas_price_middleware
        from web3.middleware.geth_poa_middleware import geth_poa_chain_id_middleware
        from web3.middleware.geth_poa_middleware import geth_poa_nonce_middleware
        from web3.middleware.geth_poa_middleware
        import logging
        logging.basicConfig(level=logging.INFO)
        logging.getLogger("urllib3").setLevel(logging.WARNING)
        logging.getLogger("requests").setLevel(logging.WARNING)
        logging.getLogger("web3").setLevel(logging.WARNING)
        logging.getLogger("ethereum").setLevel(logging.WARNING)
        logging.getLogger("urllib3").setLevel(logging.WARNING)
        logging.getLogger("requests").setLevel(logging.WARNING)
        logging.getLogger("web3").setLevel(logging.WARNING)
        logging.getLogger("ethereum").setLevel(logging.WARNING)
        logging.getLogger("urllib3").setLevel(logging.WARNING)
        logging.getLogger("requests").setLevel(logging.WARNING)
        logging.getLogger("web3").setLevel(logging.WARNING)
        logging.getLogger("ethere
        logging.getLogger("urllib3").setLevel(logging.WARNING)
        logging.getLogger("requests").setLevel(logging.WARNING)
        logging.getLogger("web3").setLevel(logging.WARNING)
        logging.getLogger("ethere
        logging.getLogger("urll

Of course, this is unrelated to what I want to do beside the function name as not all tokens are tradable on chain, and gas price is unrelated to the query and then the function is incomplete.

**Comment by MathiasReker** [comment]
Hello. You can report the bug here: https://support.github.com/contact/bug-report
